src/shared/conf-parser.c

```diff
@@ -1391,6 +1391,45 @@ int config_parse_permille(
         return 0;
 }
 
+int config_parse_io_cost_qos_scaling_percentage(
+                const char* unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        loadavg_t *io_cost_qos_scaling_percentage = data;
+        loadavg_t result;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(io_cost_qos_scaling_percentage);
+
+        r = parse_loadavg_fixed_point(rvalue, &result);
+        if (r < 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Failed to parse IOCostQOSMin= or IOCostQOSMax= value, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        /* range supported by the kernel */
+        if (LOADAVG_INT_SIDE(result) < 1 || LOADAVG_INT_SIDE(result) > 10000 ||
+                (LOADAVG_INT_SIDE(result) == 10000 && LOADAVG_DECIMAL_SIDE(result) > 0))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "IOCostQOSMin= or IOCostQOSMax= invalid");
```

Comment:
I think (not sure though) you can use `lvalue` here to get the actual key that we're trying to parse which would give a better error message.